taniumrest_connector.py

```diff
@@ -535,7 +535,12 @@ def _execute_action_support(self, param, action_result): # noqa: 901
                     "Error while parsing the 'package_parameter' field. Error: {}"
                     .format(error_message))
 
-            if len(package_parameter) != len(parameter_definition.get("parameters")):
+            count_of_params = 0
```

Comment:
```suggestion
            count_of_params = sum(1 for params in parameter_definition.get('parameters', []) if params.get('parameterType') not in TANIUMREST_PARAMETERS_WITHOUT_INPUT)
```

Reply:
@bb-splunk, I already checked the performance of your suggested solution before implemeing the code
and it was slower than given solution and more redable too.

Reply:
I believe my solution is more readable, but I understand that this is arguable. But I wonder how performance of this snippet could be reasonably measured as I'd expect it to take a relatively infinitesimal amount of time

Reply:
It's more than 145 lines so the static pipeline will fail so anyways we have to break the code which we have already done
Also, the given solution is more space and time optimized than the proposed one